core/src/avm2/multiname.rs

```diff
@@ -420,6 +404,53 @@ impl<'gc> Multiname<'gc> {
 
         AvmString::new(mc, uri)
     }
+
+    pub fn set_single_namespace(&mut self, namespace: Namespace<'gc>) {
+        self.ns = NamespaceSet::Single(namespace);
+    }
+
+    pub fn set_local_name(&mut self, name: AvmString<'gc>) {
+        self.name = Some(name);
+    }
+
+    /// Returns this multiname as a URI, panics if this multiname is malformed (has 0 namespaces)
+    pub fn as_uri(&self, mc: MutationContext<'gc, '_>) -> AvmString<'gc> {
+        let set = self.namespace_set();
+        let ns = if set.len() == 1 {
+            let first = self.namespace_set().first().expect("Malformed multiname");
+
+            match &*first.0 {
+                NamespaceData::Namespace(s) => s,
+                NamespaceData::PackageInternal(s) => s,
+                NamespaceData::Protected(s) => s,
+                NamespaceData::Explicit(s) => s,
+                NamespaceData::StaticProtected(s) => s,
+                NamespaceData::Private(s) => s,
+                NamespaceData::Any => WStr::from_units(b"*"),
+            }
+        } else {
+            WStr::empty()
+        };
+
+        if ns.is_empty() {
+            if let Some(name) = self.name {
+                return name;
+            } else {
+                // For some reason when the namespace is empty but it is an `any` name,
+                // flash gets confused and prints that the namespace is also `any`.
+                return "*::*".into();
```

Comment:
how did you reproduce this? I'm getting a different result:
```
namespace nss = "";

class C extends Proxy {
    flash_proxy override function getProperty(name: *) {
        trace(name);
        trace(name.uri);
        trace(name.uri === "");
    }
}
var a = new C();
a.nss::*; // same with a.*
```
Traces
```
*

true
```

Reply:
Ah, so this is actually left over from an incorrect test I did & I didn't double check it. 

```as
var q = new QName("*");
trace(q.uri);
trace(q.localName);
trace(q);
```
prints...
```
null
*
*::*
```
But then,
```as
var q = new QName("test");
trace(q.uri);
trace(q.localName);
trace(q);
```
prints
```

test
test
```
The weird behavior turns out not to be in `as_uri`, but in QName's constructor. When "*" is passed as the local name of the qname, it makes the namespace `any` (it would usually be public by default).